include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -301,8 +299,37 @@ class format_parse : public format_base
         stream >> value;
 
         if (stream.fail() || !stream.eof())
+        {
             throw type_conversion_failed("Argument " + in + " could not be casted to type " +
                                          get_type_name_as_string(value) + ".");
+        }
+    }
+
+    /*!\brief Tries to cast an input string into a value.
+     * \tparam option_t Must model seqan3::named_enumeration.
+     * \param[out] value Stores the cast value.
+     * \param[in]  in    The input argument to be cast.
+     *
+     * \throws seqan3::type_conversion_failed
+     */
+    template <typename option_t>
+    //!\cond
+        requires named_enumeration<option_t>
+    //!\endcond
+    void retrieve_value(option_t & value, std::string const & in)
+    {
+        auto map = seqan3::enumeration_names<option_t>;
+        auto it = map.find(in);
+
+        if (it == map.end())
```

Comment:
```suggestion
        if (auto it = map.find(in); it == map.end())
```

:nail_care:

Reply:
I `it` then available in the else scope?

Reply:
nice. it does